### PR TITLE
Added a annotation to the CR containing the CLI version.

### PR DESCRIPTION
**Pull request by olofwalker**

### CODE REVIEW

File: kubectl-cloudflow/cmd/update_docker_registry_credentials.go [hunk 1]
```diff
@@ -106,8 +106,9 @@ func createOrUpdateNamespace(k8sClient *kubernetes.Clientset, appID string) {
 func newNamespace(appID string) v1.Namespace {
 	return v1.Namespace{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:   appID,
-			Labels: domain.CreateLabels(appID),
+			Name:        appID,
+			Labels:      domain.CreateLabels(appID),
+			Annotations: domain.CreateAnnotations(version.ReleaseTag, version.BuildNumber),
```
Comment (agemooij) [inline kubectl-cloudflow/cmd/update_docker_registry_credentials.go:111]: @olofwalker Idea: we could make a difference between `created-by-cli-version` and `last-modified-by-cli-version` to add even more info.

WDYT?

Comment (agemooij) [reply kubectl-cloudflow/cmd/update_docker_registry_credentials.go:111]: I would also combine the two pieces of information into one value, e.g. `$ReleaseTag ($buildNumber)`, which is consistent with how we show it in the `kubectl cloudflow version` command output.

Comment (olofwalker) [reply kubectl-cloudflow/cmd/update_docker_registry_credentials.go:111]: I will see if we can add both those tags, we need to readback the annotations when updating the CR, sounds like a good idea.

**Comment by olofwalker** [comment]
```
k get cloudflowapplication --all-namespaces -o yaml
apiVersion: v1
items:
- apiVersion: cloudflow.lightbend.com/v1alpha1
  kind: CloudflowApplication
  metadata:
    annotations:
      com.lightbend.cloudflow/created-by-cli-version: SNAPSHOT (local build)
```

**Comment by olofwalker** [comment]
```
apiVersion: v1
items:
- apiVersion: cloudflow.lightbend.com/v1alpha1
  kind: CloudflowApplication
  metadata:
    annotations:
      com.lightbend.cloudflow/created-by-cli-version: SNAPSHOT (local build)
      com.lightbend.cloudflow/last-modified-by-cli-version: SNAPSHOT (local build)
```

**Comment by olofwalker** [comment]
Did a new test to observe any unwanted redeployment. 

Test

1. First deployed the swiss-knife application
2. Changed the buildnumber in the CLI and recompiled it
3. Started a `kubectl get pods -n swiss-knife --watch` and cleared the screen
4. Redeployed the application using the newly built CLI

Result:
- No new entries were produced by the running watch, meaning no pods were restarted by just adding a new annotation to the CR.
- The annotations had been added to the application.

```
      com.lightbend.cloudflow/created-by-cli-version: SNAPSHOT (local build)
      com.lightbend.cloudflow/last-modified-by-cli-version: SNAPSHOT (local build
        - new build of the cli)
```

File: kubectl-cloudflow/domain/cloudflowapplication_test.go [hunk 2]
```diff
@@ -53,15 +53,20 @@ func Test_validateOwnerReferenceGeneration(t *testing.T) {
...
 	assert.True(t, ownerReference.Kind == "CloudflowApplication")
 	assert.True(t, ownerReference.APIVersion == "cloudflow.lightbend.com/v1alpha1")
 	assert.True(t, ownerReference.Name == "sensor-data-scala")
 	assert.True(t, *ownerReference.Controller == true)
 	assert.True(t, *ownerReference.BlockOwnerDeletion == true)
 
+	// Test updating the annotation
+	cr = UpdateCloudflowApplication(cr.Spec, cr, "v1.3.2", "35ac352")
+	assert.True(t, cr.GetObjectMeta().GetAnnotations()["com.lightbend.cloudflow/created-by-cli-version"] == "v1.3.1 (34ab342)")
+	assert.True(t, cr.GetObjectMeta().GetAnnotations()["com.lightbend.cloudflow/last-modified-by-cli-version"] == "v1.3.2 (35ac352)")
```
Comment (maasg) [inline kubectl-cloudflow/domain/cloudflowapplication_test.go:69]: :heavy_check_mark:

File: kubectl-cloudflow/domain/cloudflowapplication.go [hunk 3]
```diff
@@ -194,36 +194,38 @@ func ApiVersion() string {
 }
 
 // UpdateCloudflowApplication creates a CloudflowApplication struct that can be used with a Update call
-func UpdateCloudflowApplication(spec CloudflowApplicationSpec, resourceVersion string) CloudflowApplication {
+func UpdateCloudflowApplication(spec CloudflowApplicationSpec, currentAppCR CloudflowApplication, releaseTag string, buildNumber string) CloudflowApplication {
```
Comment (maasg) [inline kubectl-cloudflow/domain/cloudflowapplication.go:197]: for consistency, `currentAppCR` could be called `currentApp` xor `UpdateCloudflowApplication` should be `UpdateCloudflowApplicationCR` but I think overall it's called `CloudflowApplication` so I'd go with the former.

**Pull request merged**